### ISSUE TITLE
An administrator will be able to view user chats to monitor

### ISSUE THREAD

**Issue by Mordechay0000** [NodeBB/NodeBB#11856]
### Description

An administrator will be able to view user chats to monitor, by logging in to https://forum-dev.com/user/username/chats

**Comment by julianlam** [comment]
I think this may be a privacy violation.

**Comment by Mordechay0000** [comment]
You're right I thought about it and that's why I wrote that it's for monitoring, you can add it in the privacy terms or warn the user before opening a chat, there must be an option to monitor the chats, if a user complains about another user needs to be able to check it, there's already a plugin that someone built that does it but I think it's more appropriate for it to be built-in

**Comment by tankerkiller125** [comment]
I feel like the correct way to handle reports in chats would be to send the last say 4 messages and the one reported (2 before, reported message, 2 after if they exist) to the admin and nothing else. An admin shouldn't have access to chats at all except the ones they're invited to.

**Comment by oplik0** [comment]
This is honestly what I'd consider a good compromise that could be adapted even for e2ee chats (reporting client submits the few decrypted messages)

I'm not sure about the number of messages for context - perhaps something like minimum word count (that is, full messages selected until a specified word count is reached) would be more useful (but I don't have much experience with moderating with such limited context, so maybe just number of messages would be enough while being simpler both code-wise and for users to understand).

**Comment by Mordechay0000** [comment]
There is currently no report button at all
By the way, because messages are not always encrypted, the forum moderator has access to chats, there is a plugin that does this

**Comment by Mordechay0000** [comment]
Now there's no report button at all...
By the way, because the messages are not encrypted and the administrator can view them, there is a plugin that does this

**Comment by ShlomoCode** [comment]
> send the last say 4 messages

In Google Chat, report a user to Google who sends 50 recent messages
This **may** be a measure of what is accepted in the industry

**Comment by julianlam** [comment]
It's hard to say how many messages you should send in a report. 4 messages may be adequate for some scenarios not others.

I was thinking we could send all messages during that _session_, but even that doesn't work if an argument spans multiple days, etc.

**Comment by tankerkiller125** [comment]
I think 50 is far too many, not surprising for a privacy invading company like Google.

I think maybe a middle ground of like 25 might be enough for the vast majority of cases. And for the cases where it's not maybe have a button on the admin side where they can request an additional 25 messages from the person who made the initial report? I don't know how feasible that would be, but it seems like a reasonable middle ground that can still provide enough context will preserving more privacy by default.

**Comment by ShlomoCode** [comment]
> I think 50 is far too many, not surprising for a privacy invading company like Google.

It should be remembered that the person who violates privacy is not Google/the forum administrators, but the person who decided to report,
He could have just as well sent screenshots
I agree that 50 sounds a lot, but 4 sounds too little.
It is possible to create a combination of amount of messages + amount of time, that is, if there was a conversation with many messages in one day, everything will be sent, but if there was a conversation over a week, let's say only 25 recent messages will be sent.

**Comment by julianlam** [comment]
@ShlomoCode That's what I originally thought we could do, because we could signify any "gap" as an indicator of a new conversation. However that assumes that a chat interaction takes place in real-time, which isn't necessarily a guarantee. Someone could send someone else a spam message, and the recipient might not check that message until a couple days later, etc.

So... it is hard to say :sweat_smile:

**Comment by ShlomoCode** [comment]
@julianlam Even in such a case, according to my proposal, a report will be sent, just not the long-term history, but a limited number of messages.

**Comment by mySebbe** [comment]
There is still no core chat-report path, and adding an administrator bypass to all rooms would implement the broad surveillance model that maintainers objected to. Should the acceptable scope instead be a user-initiated report containing a fixed, immutable window of messages around the reported message? If so, we still need the window size, who can view it, whether participants are notified, retention/deletion rules, and an audit log. I would not prepare a PR until that privacy contract is explicit.

**Comment by barisusakli** [comment]
https://github.com/palmoni5/nodebb-plugin-admin-chats
